lib/util.js

```diff
@@ -128,6 +128,30 @@ exports.chooseFormat = (formats, options) => {
       format = formats[formats.length - 1];
       break;
 
+    case 'highestaudio':
+      formats = exports.filterFormats(formats, 'audio');
+      format = formats[0];
+      for (let f of formats) {
+        if (f.audioBitrate > format.audioBitrate
+          || (f.audioBitrate === format.audioBitrate && format.encoding && !f.encoding))
+            format = f;
+      }
+      break;
+
+    case 'highestvideo':
+      function getBitrate(f) {
+        let s = f.bitrate.split('-');
+        return parseFloat(s[s.length - 1], 10);
+      }
+      formats = exports.filterFormats(formats, 'video');
+      format = formats[0];
+      for (let f of formats) {
+        if (getBitrate(f) > getBitrate(format)
+          || (getBitrate(f) === getBitrate(format) && format.audioEncoding && !f.audioEncoding))
+            format = f;
```

Comment:
This line is currently not being test covered: https://codecov.io/gh/fent/node-ytdl-core/compare/692da6cc6d7206f8149382186742fa98299eb2ec...9fbcf14693ae97c03a327596f7ab191cb1dc587f/diff

Can be fixed by either
* setting the above `format`s to `format = null;` and then adding another check in the loops for `!format`
* changing the order of `formats` in `util-test.js` so that the format at the top of the list is not the one we initially want, by calling `.reverse()` on it for example